src/main/java/pl/tomaszdziurko/jvm_bloggers/view/admin/blogs/BlogActionPanel.java

```diff
@@ -0,0 +1,74 @@
+package pl.tomaszdziurko.jvm_bloggers.view.admin.blogs;
+
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.markup.html.form.AjaxButton;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.markup.html.panel.Panel;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.spring.injection.annot.SpringBean;
+
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.Blog;
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.BlogRepository;
+import pl.tomaszdziurko.jvm_bloggers.view.panels.CustomFeedbackPanel;
+
+import java.io.Serializable;
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+
+public class BlogActionPanel extends Panel {
+
+    @SpringBean
+    private BlogRepository blogRepository;
+
+    private final Form<?> form;
+    private final IModel<Blog> blogModel;
+    private final CustomFeedbackPanel feedbackPanel;
+
+    public BlogActionPanel(
+        String id, Form<?> form, IModel<Blog> blogModel, CustomFeedbackPanel feedbackPanel) {
+
+        super(id);
+        this.form = form;
+        this.blogModel = blogModel;
+        this.feedbackPanel = feedbackPanel;
+
+        add(createActivateBlogButton());
+        add(createDeactivateBlogButton());
+    }
+
+    private AjaxButton createActivateBlogButton() {
+        AjaxButton activate = createBlogActionButton("activateBlog",
+            (Consumer<Blog> & Serializable) b -> b.setActive(true),
+            format("'%s' blog activated", blogModel.getObject().getAuthor()));
+        activate.setVisible(!blogModel.getObject().isActive());
+        return activate;
+    }
+
+    private AjaxButton createDeactivateBlogButton() {
+        AjaxButton deactivate = createBlogActionButton("deactivateBlog",
+            (Consumer<Blog> & Serializable) b -> b.setActive(false),
+            format("'%s' blog deactivated", blogModel.getObject().getAuthor()));
+        deactivate.setVisible(blogModel.getObject().isActive());
+        return deactivate;
+    }
+
+    private AjaxButton createBlogActionButton(
```

Comment:
These two lines cries for reformatting: either move some stuff from 2nd line to the 1st one or ideally place each parameter in a separate line to improve readability.